### PR TITLE
feat: Add support for multiple exchange instances

### LINKED ISSUE #14: Support Multiple Exchange Instances for Rocket Server Integration
### Description:

In order to integrate the exchange with the Rocket server and satisfy its requirements, we need to add functionality to create multiple instances of the exchange. This will allow the server to handle requests for different exchange instances.

### Tasks:

1. Write the functionality to support multiple exchange instances.
2. Ensure proper synchronization and resource management for multiple instances.
3. Update test cases to cover the new functionality and ensure there are no regressions.

### PR DESCRIPTION
**Pull request by DieHard073055**
- Introduce a unique ID for each Exchange instance
- Add `Exchanges` struct to manage multiple instances
- Update `Exchange::with_capital` and `Exchange::with_price_feed` to use mutable references
- Add test for generating UUIDs based on order details
- Add `uuid` crate with `v4` and `fast-rng` features

This change allows creating and managing multiple instances of the Exchange, which can be useful for running simulations or managing multiple user accounts in a trading application.

### CODE REVIEW

File: src/exchange/exchange.rs [hunk 1]
```diff
@@ -356,15 +356,18 @@ mod test {
 
     #[test]
     fn test_initialize_exchange_with_capital() {
-        let ex = Exchange::new().with_capital(vec![(String::from("BTC"), dec!(12.0))]);
+        let mut ex = Exchange::new();
+        ex.with_capital(vec![(String::from("BTC"), dec!(12.0))]);
```
Comment (pamdayne) [inline src/exchange/exchange.rs:360]: Nice clean up

File: src/lib.rs [hunk 2]
```diff
@@ -1,8 +1,192 @@
 pub mod exchange;
 
+/*
+TODO:
+write unit tests for the code in this file.
+something else is code coverage not related to the file.
+get the PR done asap. Next week is the next project.
+
+ */
```
Comment (pamdayne) [inline src/lib.rs]: Just checking: the tests found in this file is not related to this comment?

File: src/lib.rs [hunk 3]
```diff
@@ -1,8 +1,192 @@
...
+        exchange.with_capital(vec![(symbol.to_string(), amount)]);
+        Ok(())
+    }
+    pub fn add_price_feed(
+        &mut self,
+        instance_id: &str,
+        symbol: &str,
+        interval: &str,
+        limit: i32,
+    ) -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
```
Comment (pamdayne) [inline src/lib.rs:70]: is the single open quote `'` at `static` - is that normal?

**Review by pamdayne — COMMENTED** [review]
Just keep in mind with a few Github actions warning for unused imports.

File: src/lib.rs (hunk 2 above)
Comment (DieHard073055) [reply src/lib.rs]: Yea, ill remove it

File: src/lib.rs (hunk 3 above)
Comment (DieHard073055) [reply src/lib.rs:70]: Its rust's explicit lifetime annotations.
https://doc.rust-lang.org/rust-by-example/scope/lifetime/explicit.html

Comment (pamdayne) [reply src/lib.rs:70]: gotcha - thanks!

**Review by pamdayne — APPROVED** [review]
